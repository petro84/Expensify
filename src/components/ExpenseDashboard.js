import React from 'react';

import ExpenseSummary from './ExpenseSummary';
import ExpenseList from './ExpenseList';
import ExpenseListFilters from './ExpenseListFilters';

const ExpenseDashboard = () => (
  <div>
    <ExpenseSummary />
    <ExpenseListFilters />
    <ExpenseList />
  </div>
);

export default ExpenseDashboard;
