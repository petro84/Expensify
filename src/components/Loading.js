import React from 'react';
import loader from '../images/loader.gif';

const Loading = () => (
    <div className='loader'>
        <img className='loader__image' src={loader} />
    </div>
);

export default Loading;