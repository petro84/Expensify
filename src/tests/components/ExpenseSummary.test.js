import React from 'react';
import { shallow } from 'enzyme';

import { ExpenseSummary } from '../../components/ExpenseSummary';

test('should show total for single expense', () => {
  const wrapper = shallow(<ExpenseSummary expenseCount={1} expensesTotal={195} />);
  expect(wrapper).toMatchSnapshot();
});

test('should show total of multiple expenses', () => {
  const wrapper = shallow(<ExpenseSummary expenseCount={3} expensesTotal={114195} />);
  expect(wrapper).toMatchSnapshot();
});
