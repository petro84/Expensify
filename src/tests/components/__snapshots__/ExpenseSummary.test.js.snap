// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should show total for single expense 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totalling 
    $1.95
    .
  </h1>
</div>
`;

exports[`should show total of multiple expenses 1`] = `
<div>
  <h1>
    Viewing 
    3
     
    expenses
     totalling 
    $1,141.95
    .
  </h1>
</div>
`;
